state-sync/storage-service/server/src/lib.rs

```diff
@@ -356,6 +356,85 @@ impl StorageReader {
     pub fn new(storage: Arc<dyn DbReader<DpnProto>>) -> Self {
         Self { storage }
     }
+
+    /// Returns the account states range held in the database (lowest to highest).
+    /// Note: it is currently assumed that if a node contains a transaction at a
+    /// version, V, the node also contains all account states at V.
+    fn fetch_account_states_range(
+        &self,
+        latest_version: Version,
+        transactions_range: &Option<CompleteDataRange<Version>>,
+    ) -> Result<Option<CompleteDataRange<Version>>, Error> {
+        let pruning_window = self
+            .storage
+            .get_state_prune_window()
+            .map(|window| window as u64);
+        if let Some(pruning_window) = pruning_window {
+            if latest_version > pruning_window {
+                // lowest_account_version = latest_version - pruning_window + 1;
+                let mut lowest_account_version =
+                    latest_version.checked_sub(pruning_window).ok_or_else(|| {
+                        Error::UnexpectedErrorEncountered(
+                            "Lowest account states version has overflown!".into(),
+                        )
+                    })?;
+                lowest_account_version =
+                    lowest_account_version.checked_add(1).ok_or_else(|| {
+                        Error::UnexpectedErrorEncountered(
+                            "Lowest account states version has overflown!".into(),
+                        )
+                    })?;
+
+                // Create the account range
+                let account_range = CompleteDataRange::new(lowest_account_version, latest_version)
+                    .map_err(|error| Error::UnexpectedErrorEncountered(error.to_string()))?;
+                Ok(Some(account_range))
+            } else {
+                // The pruner hasn't kicked in yet, so return the transactions range
+                Ok(*transactions_range)
+            }
+        } else {
+            // There is no pruning window, return the transactions range
+            Ok(*transactions_range)
+        }
```

Comment:
You can drop this to the bottom, and then just add `return Ok(Some(account_range))` to line 391.

Then have a comment, 
```
// Not pruned, return the transactions range
```

Reply:
Done!